trussworks/react-uswds
# fix: add ActionType.CLEAR_FOCUS to indicate a user has left the field

# Summary
The ComboBox has issues with "losing" its focus when clicking outside of the component.

https://user-images.githubusercontent.com/8367504/108426115-35955180-71f0-11eb-91c1-9ea40d8bd920.mov

A related issue is that the `FocusMode` is still set to `FocusMode.Input` even after the user has left the component.

This PR allows the user to:
1. Unfocus from the ComboBox when they click outside
2. Resets the `FocusMode` to `None` when they click outside

I achieved this by creating a new `ActionType` called `CLEAR_FOCUS` to indicate when a user has clicked outside of the combo box. I am open to suggestions on the name. This action is different from `ActionType.CLEAR` because `CLEAR` empties the input, whereas `CLEAR_FOCUS` indicates the user has left the field.

## Related Issues or PRs
Fixes #901 

## How To Test
1. Open Storybook
2. Focus ComboBox by opening its options or typing in the input
3. Click outside of the component and the input should lose focus.

### Screenshots (optional)

After Fix:

https://user-images.githubusercontent.com/8367504/108428519-89556a00-71f3-11eb-8029-d19cf7703133.mov
